tests/unit/aiplatform/test_jobs.py

```diff
@@ -969,3 +977,93 @@ def test_batch_predict_job_with_versioned_model(
+        create_mdm_job_mock.return_value = (
+            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
+                name=_TEST_MDM_JOB_NAME,
+                display_name=_TEST_DISPLAY_NAME,
+                state=_TEST_JOB_STATE_RUNNING,
+            )
+        )
+        yield create_mdm_job_mock
+
+
+@pytest.fixture
+def get_mdm_job_mock():
+    with mock.patch.object(
+        _TEST_API_CLIENT, "get_model_deployment_monitoring_job"
+    ) as get_mdm_job_mock:
+        get_mdm_job_mock.return_value = (
+            gca_model_deployment_monitoring_job_compat.ModelDeploymentMonitoringJob(
+                name=_TEST_MDM_JOB_NAME,
+                display_name=_TEST_DISPLAY_NAME,
+                state=_TEST_JOB_STATE_RUNNING,
+                endpoint=_TEST_ENDPOINT,
+            )
+        )
+        yield get_mdm_job_mock
+
+
+@pytest.fixture
+def mock_deployed_model():
+    with mock.patch.object(gca_endpoint_compat, "DeployedModel") as mock_deployed_model:
+        mock_deployed_model.id = _TEST_ID
+        mock_deployed_model.explanation_spec.parameters = "TEST_VALUE"
+        yield mock_deployed_model
+
+
+@pytest.fixture
+def endpoint_list_models_mock(mock_deployed_model):
+    with mock.patch.object(
+        aiplatform.Endpoint, "list_models"
+    ) as endpoint_list_models_mock:
+        endpoint_list_models_mock.return_value = [mock_deployed_model]
+        yield endpoint_list_models_mock
+
+
+@pytest.fixture
+def update_mdm_job_mock(get_mdm_job_mock, endpoint_list_models_mock):
+    with mock.patch.object(
+        job_service_client.JobServiceClient, "update_model_deployment_monitoring_job"
+    ) as update_mdm_job_mock:
+        mock_client_call = mock.Mock(
+            spec=job_service_client.JobServiceClient.update_model_deployment_monitoring_job
+        )
+        update_mdm_job_mock.return_value = mock_client_call(
+            model_deployment_monitoring_job=get_mdm_job_mock,
+            update_mask=field_mask_pb2.FieldMask(
+                paths=["model_deployment_monitoring_objective_configs"]
+            ),
+        )
+        yield update_mdm_job_mock
+
+
+@pytest.mark.usefixtures("google_auth_mock")
+class TestModelDeploymentMonitoringJob:
+    def setup_method(self):
+        reload(initializer)
+        reload(aiplatform)
+
+    def teardown_method(self):
+        initializer.global_pool.shutdown(wait=True)
+
+    @pytest.mark.usefixtures(
+        "create_mdm_job_mock", "get_mdm_job_mock", "update_mdm_job_mock"
+    )
+    def test_update_mdm_job(self):
+        job = jobs.ModelDeploymentMonitoringJob(
+            model_deployment_monitoring_job_name=_TEST_MDM_JOB_NAME
+        )
+        new_config = aiplatform.model_monitoring.ObjectiveConfig(
+            explanation_config=aiplatform.model_monitoring.ExplanationConfig()
+        )
+        job.update(objective_configs=new_config)
+        assert job._gca_resource.model_deployment_monitoring_objective_configs[
```

Comment:
This test should also assert all client requests were made as expected.

Reply:
Hmm somehow when I tried to call `update_mdm_job_mock.assert_called_once...`, I got an attribute error saying `AttributeError: 'function' object has no attribute 'assert_called_once'`. Not sure what I did wrong here because it looks like the fixture is set up similarly as other fixtures in this test file.

Reply:
update: I have fixed this issue. It looks like marking the use of a fixture isn't enough to import all the context & I needed to include the fixture as an argument in the test function too.